game/transfers.py

```diff
@@ -166,21 +166,46 @@ def find_escape_route(self) -> Optional[ControlPoint]:
             return self.transport.find_escape_route()
         return None
 
-    def proceed(self) -> None:
-        if not self.destination.is_friendly(self.player):
-            logging.info(f"Transfer destination {self.destination} was captured.")
-            if self.position.is_friendly(self.player):
-                self.disband_at(self.position)
-            elif (escape_route := self.find_escape_route()) is not None:
-                self.disband_at(escape_route)
+    def disband(self) -> None:
+        if self.position.is_friendly(self.player):
+            self.disband_at(self.position)
+        elif (escape_route := self.find_escape_route()) is not None:
+            self.disband_at(escape_route)
+        else:
+            logging.info(
+                f"No escape route available. Units were surrounded and destroyed "
+                "during transfer."
+            )
+            self.kill_all()
+
+    def is_completable(self, network: TransitNetwork) -> bool:
+
+        if not self.position.is_friendly(self.player):
+            logging.info(f"Current transfer position {self.position} was captured.")
+            return False
```

Comment:
originally it was never handled thats why i made this PR :) the handling was just checking for the destination rather than checking the path.
This led to a `ValueError` as `shortest_path_with_cost` checked if the `origin` and `destination` are in the **same network**. which they are not as origin was captured -> Thats also the answer to the question above what was the intention of this bug fix.

I will have a look on this one again. At least then it will reach 100% perfectness :)

Reply:
i de-complicated the logic and now only check if the processing can be fullfilled for already processing transfers by checking if the next_stop is friendly. For unplanned transfers i only check if the transit network offers us a route between current position and destination as well as checking if the current position they're halting at is friendly

Reply:
> originally it was never handled

Is that right? I think you're right about there being a bug for a cheat capture, but I think a regular capture worked?

The base is captured in `Event.commit_captures()`. Units that are _stationed_ (not in a transfer) at that base are retreated or captured. `Game.pass_turn` runs after, which eventually calls `Coaltiion.end_turn`, which calls `PendingTransfers.perform_tansfers`, which calls `TransferOrder.proceed`. The transfer is only disbanded if the _destination_ was captured. If the current location was captured it will still proceed.

The changes to check whether the path is completable a good addition 👍